bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,252 @@
+                        sprintf(
+                            "Content type with identifier '%s' does not exist: %s",
+                            $allowedContentType,
+                            $exception->getMessage(),
+                        ),
+                    );
+
+                    continue;
+                }
+            }
+        }
+
+        $this->applyContentTypeLimit($query, $contentTypeIds);
+
+        $pager = $this->getPager($query, $contentTypeIds);
+
+        if ($pager->getNbResults() === 0) {
+            $output->writeln('No content found.');
+
+            return Command::FAILURE;
+        }
+
+        $limit = $input->getOption('limit');
+        $offset = 0;
+
+        $this->style->createProgressBar($pager->getNbResults());
+        $this->style->progressStart();
+
+        $results = $pager->getAdapter()->getSlice($offset, $limit);
+        while (count($results) > 0) {
+            $this->processResults($results);
+            $offset += $limit;
+            $results = $pager->getAdapter()->getSlice($offset, $limit);
+        }
+
+        $this->style->progressFinish();
+
+        $this->style->info('Done.');
+
+        return Command::SUCCESS;
+    }
+
+    private function processResults(array $results): void
+    {
+        foreach ($results as $result) {
+            try {
+                $languageId = $result['initial_language_id'];
+                $language = $this->loadLanguage($languageId);
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Language with id #%d does not exist: %s',
+                        $languageId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
+            }
+
+            try {
+                $contentId = $result['id'];
+
+                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                $content = $this->repository->sudo(
+                    fn () => $this->repository->getContentService()->loadContent(
+                        $contentId,
+                        [$language->getLanguageCode()],
+                    ),
+                );
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Content with id #%d does not exist: %s',
+                        $contentId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
+            }
+
+            try {
+                $action = $this->scheduledVisibilityService->updateVisibilityIfNeeded($content);
+            } catch (InvalidStateException $exception) {
+                $this->logger->error($exception->getMessage());
+
+                continue;
+            }
+
+            if ($action !== VisibilityUpdateResult::NoChange) {
```

Comment:
Let's put logging back to `ScheduledVisibilityService`. Outside of it, we should only catch the exception and log an error.

Reply:
fixed in b325c484488f68c043f8249a60b0d515697f35f5